src/playground/public/assets/js/app.ts

```diff
@@ -29,12 +29,18 @@ async function printAskVersion(elemId: string, askScriptServerUrl: string) {
 function registerAskScriptEditor(
   editorElementId: string,
   runElementId: string,
-  askScriptServerUrl: string
+  askScriptServerUrl: string,
+  dirty: boolean
 ) {
   const editor = ace.edit(editorElementId);
   editor.setTheme('ace/theme/twilight');
   editor.session.setMode('ace/mode/javascript');
   editor.session.setOption('useWorker', false);
+  editor.session.on('change', function () {
+    if (dirty) return;
+    confirmExitFromPlayground();
+    dirty = true;
```

Comment:
There is no counterpart that would set `dirty` to false so we'll require confirmation from everyone once they have made any edit. Given that the purpose of the playground is to execute the inputed code maybe it would make sense to only require it if the code wasn't executed?

Reply:
I stay with this solution, now the flag dirty change if we  run a script in the playground